### PR TITLE
send new user notification email if enabled

### PR DESCRIPTION
**Pull request by Schparky**
Following is HR's request:

> An OU HR contact needs to receive a notification email when one of their staff gets a new SIL IdP account. 
>
> In some OUs, the HR contact will forward the email to their OU IT, who will use this information to verify with Sherlock that the user has properly set up their 2SV, password, etc.
>
> The notification email can be similar to the kind of email that HR receives when a Google account is created for one of their staff.
>
> Two new columns will be added to the [_Workday_] Integration report so that the SIL IdP integration can know who the appropriate HR contact is (based on the HR admin of the staff's Company of assignment).
>
> - HR Contact Name
> - HR Contact Email
>
> These columns would supply the HR contact name and email address for the notification email to be sent to HR.

### CODE REVIEW

File: application/features/notification.feature [hunk 1]
```diff
@@ -6,8 +6,29 @@ Feature: Sending notifications
     When I call the sendMissingEmailNotice function
     Then an email is sent
 
+  Scenario: Don't send missing email notification
+    Given a specific user exists in the ID Store
+      But the user does not exist in the ID Broker
+    When I get the user info from the ID Store and send it to the ID Broker
+    Then an email with subject "Email address missing" is not sent
```
Comment (forevermatt) [inline application/features/notification.feature:13]: This scenario seems to leave something implied that seems important to its statement about what the application does. Is it important, for instance, that the user info obtained from the ID Store (and that the application will try to send to the ID Broker) lacks an email address?

Comment (forevermatt) [reply application/features/notification.feature:13]: I guess I'm just confused on why this test was added, and what it's proving about the code.

File: application/features/bootstrap/SyncContext.php [hunk 2]
```diff
@@ -28,18 +29,21 @@ class SyncContext implements Context
     private $idBroker;
 
     /** @var FakeIdStore */
-    private $idStore;
+    protected $idStore;
```
Comment (forevermatt) [inline application/features/bootstrap/SyncContext.php:32]: Why the visibility change?

**Review by forevermatt — COMMENTED** [review]
Overall this looks good, though I have a couple of questions.

File: application/features/notification.feature (hunk 1 above)
Comment (Schparky) [reply application/features/notification.feature:13]: It's the "normal" case, proving that you won't send a "missing email" message for every new user. I'll see if I can reword it to help with that understanding.

File: application/features/bootstrap/SyncContext.php (hunk 2 above)
Comment (Schparky) [reply application/features/bootstrap/SyncContext.php:32]: `NotificationContext` inherits `SyncContext` and uses `idStore` to get the user that was added earlier in the scenario. I could have changed the implementation to store the user as a property instead, but kept with the existing implementation.

File: application/features/notification.feature (hunk 1 above)
Comment (forevermatt) [reply application/features/notification.feature]: Ahh, thanks. Maybe just rewording the title of the scenario would accomplish that.

**Review by forevermatt — APPROVED** [review]
